### PR TITLE
refactor: use binary-matching syntax for scatter-add impl

### PR DESCRIPTION
**Pull request by polvalente**
As discussed in https://github.com/elixir-nx/nx/pull/458#discussion_r711698612

### CODE REVIEW

**Review by josevalim — APPROVED** [review]
LGTM! It is your call if you want to merge this or not!

**Comment by polvalente** [comment]
I'm gonna apply a fix that appeared in #466, and merge eveything afterwards :)

**Pull request merged**
